ingestion/create_derived_table_domains_source/source.py

```diff
@@ -0,0 +1,90 @@
+import json
+from typing import Dict, Iterable
+
+import boto3
+import datahub.emitter.mce_builder as builder
+from botocore.exceptions import ClientError, NoCredentialsError
+from datahub.emitter.mcp import MetadataChangeProposalWrapper
+from datahub.ingestion.api.common import PipelineContext
+from datahub.ingestion.api.decorators import config_class
+from datahub.ingestion.api.source import Source, SourceReport
+from datahub.ingestion.api.workunit import MetadataWorkUnit
+from datahub.metadata.schema_classes import ChangeTypeClass, DomainPropertiesClass
+
+from ingestion.create_derived_table_domains_source.config import (
+    CreateDerivedTableDomainsConfig,
+)
+
+
+@config_class(CreateDerivedTableDomainsConfig)
+class CreateDerivedTableDomains(Source):
+    source_config: CreateDerivedTableDomainsConfig
+    report: SourceReport = SourceReport()
+
+    def __init__(self, config: CreateDerivedTableDomainsConfig, ctx: PipelineContext):
+        super().__init__(ctx)
+        self.source_config = config
+
+    @classmethod
+    def create(cls, config_dict, ctx):
+        config = CreateDerivedTableDomainsConfig.parse_obj(config_dict)
+        return cls(config, ctx)
+
+    def get_workunits(self) -> Iterable[MetadataWorkUnit]:
+        manifest = self._get_manifest(self.source_config.manifest_s3_uri)
+        for domain_name in self._get_domains(manifest):
+            mcp = self._make_domain(domain_name)
+            wu = MetadataWorkUnit("single_mcp", mcp=mcp)
+            self.report.report_workunit(wu)
+
+            yield wu
+
+    def _get_domains(self, manifest) -> set[str]:
+        return set(
+            manifest["nodes"][node]["fqn"][1]
```

Comment:
That would be more readable and I considered it, but I left it so that no alterations were made to the CaDeT definitions. Another option is to transform the domain name on the front end side.

Reply:
Yep, we can do that 👍🏻